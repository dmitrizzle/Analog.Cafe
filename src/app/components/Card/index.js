// tools
import React from "react"

// styles
import {
  Card as CardElement,
  CardButton,
  CardCaption,
  CardHeader
} from "./styles"

// return
export const Card = props => {
  return (
    <CardElement style={props.style}>
      <CardHeader title={props.error && props.error}>
        <h3 onClick={event => event.stopPropagation()}>{props.title}</h3>
        {// Stubborn popup shows no close button but requires buttons within
        !(
          props.stubborn &&
          props.buttons &&
          Object.keys(props.buttons).length !== 0
        ) ? (
          <a href="#close" onClick={event => event.preventDefault()}>
            âœ•
          </a>
        ) : null}
      </CardHeader>
      <figure onClick={event => event.stopPropagation()}>
        <img
          src={
            props.image ||
            "data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
          }
          style={{
            display: typeof props.image === "undefined" ? "none" : "initial"
          }}
          alt="Card"
        />
        <figcaption>
          <CardCaption
            style={{ padding: typeof props.text === "undefined" ? "0" : "" }}
          >
            {props.text}
          </CardCaption>
        </figcaption>
      </figure>
      {props.buttons &&
        Object.keys(props.buttons).length !== 0 &&
        props.buttons.map(function(button, i) {
          return (
            <CardButton
              onClick={button.onClick}
              to={button.to}
              key={button.to}
              red={button.red ? true : null}
            >
              {button.text}
            </CardButton>
          )
        })}
    </CardElement>
  )
}

export { CardModal, ModalOverlay } from "./components/Modal"
export { default as CardEditableProfile } from "./components/EditableProfile"
