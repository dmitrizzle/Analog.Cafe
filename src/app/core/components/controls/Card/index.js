import React from "react"

import ButtonGroupDivider from "../Button/components/ButtonGroupDivider"
import ButtonKeyword from "../Button/components/ButtonKeyword"
import CardButton from "./components/CardButton"
import CardFigure from "./components/CardFigure"
import CardHeader from "./components/CardHeader"
import CardPopup from "./components/CardPopup"
import FollowButtons from "../ArticleActions/components/FollowButtons"
import Search from "../../forms/Search"
import Spinner from "../../icons/Spinner"

export default class extends React.PureComponent {
  constructor(props) {
    super(props)
    this.state = {
      // searchMode: false,
      searchText: ""
    }
  }
  // handleSearchMode = searchMode => {
  //   this.setState({
  //     searchMode
  //   });
  // };
  handleSearchText = searchText => {
    this.setState({ searchText })
  }
  // componentWillReceiveProps = () => {
  //   this.setState({
  //     searchMode: false
  //   });
  // };

  render = () => {
    return (
      <CardPopup style={this.props.style} id={this.props.id}>
        {!this.props.headless && (
          <CardHeader
            error={this.props.error}
            stubborn={this.props.stubborn}
            buttons={this.props.buttons}
            title={this.props.title}
            noStar={this.props.menu || this.props.search}
          />
        )}
        <CardFigure image={this.props.image} text={this.props.text} />
        {this.props.search && [
          <Search
            onClick={event => event.stopPropagation()}
            formLocation={this.props.searchFormLocation}
            // searchMode={this.handleSearchMode}
            // stateOverwrite={this.state.searchMode}
            key="search"
            searchText={this.handleSearchText}
            menu={this.props.menu}
          />
        ]}
        {this.props.buttons &&
          // !this.state.searchMode &&
          Object.keys(this.props.buttons).length !== 0 &&
          this.props.buttons.map(function(button, i) {
            let keyword, buttonText
            if (button && button.text && typeof button.text === "string") {
              const keywordMatch = button.text.match(/\[(.*?)\]/)
              keyword = keywordMatch ? keywordMatch[1] : null
              buttonText = button.text.replace(`[${keyword}]`, "")
            }
            if (button && button.text && React.isValidElement(button.text)) {
              buttonText = button.text
            }
            return button && button.to && button.text ? (
              <CardButton
                to={button.to}
                key={button.to}
                onClick={button.onClick}
                branded={button.branded ? true : null}
                inverse={button.inverse ? true : null}
                mobile={button.mobile ? button.mobile : null}
                animationUnfold={button.animationUnfold}
              >
                {button.loading && <Spinner />}
                {buttonText}
                {keyword && (
                  <ButtonKeyword
                    branded={button.branded}
                    inverse={button.inverse}
                  >
                    {keyword}
                  </ButtonKeyword>
                )}
              </CardButton>
            ) : button && button.divider ? (
              <ButtonGroupDivider key={i} />
            ) : null
          })}
        {this.props.socialButtons && <FollowButtons />}
      </CardPopup>
    )
  }
}
