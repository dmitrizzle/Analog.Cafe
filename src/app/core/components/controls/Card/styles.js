import styled, { css } from "styled-components"

import { LinkButton } from "../Button"
import { Sidenote } from "../../styles/CaptionStyles"
import EMOJI from "../../../constants/EMOJI"

// css
export const CardButtonStyles = css`
  max-width: 100%;
  margin: 0;
  border-radius: 0;
  &:active {
    box-shadow: 0 -1px 0 ${props => props.theme.color.foreground()};
  }
  ${props =>
    props.responsiveMobileOnly &&
    props.theme.size.breakpoint.min.l`display:none;`};
`
export const Card = styled.div`
  position: relative;
  display: block;
  background: ${props => props.theme.color.background()};
  overflow: hidden;
  max-width: ${props => props.theme.size.breakpoint.stops.min}px;

  border-radius: ${props => props.theme.effects.borderRadius.med}em;
  box-shadow: 0 ${props => props.theme.size.block.spacing / 2}em
    ${props => props.theme.size.block.spacing * 2}em
    ${props => props.theme.color.foreground(props.theme.opacity.half)};

  ${props => props.theme.size.breakpoint.max.xs`
		border-radius: 0;
	`} figure {
    margin: 0;
    border-bottom: ${props => props.theme.elements.thickBorder};
    img {
      width: 100%;
    }
  }
  & > button,
  & > a {
    ${CardButtonStyles};
  }
`
export const CardFlattened = styled(Card)`
  margin: ${props => props.theme.size.block.padding}em auto;
  box-shadow: 0 1px 1px
      ${props => props.theme.color.foreground(props.theme.opacity.least * 2)},
    0 0 0 1px
      ${props => props.theme.color.foreground(props.theme.opacity.least)};
  border-radius: ${props => props.theme.effects.borderRadius.small}em;

  ${props => props.theme.size.breakpoint.max.xs`
		border-radius: 0;
    width: 100vw;
    margin-left: -${props => props.theme.size.block.padding}em;
	`};
`
export const CardButton = styled(LinkButton)`
  ${CardButtonStyles};
`

export const CardCaptionStyles = css`
  text-align: left;
  padding: ${props => props.theme.size.block.spacing}em
    ${props => props.theme.size.block.padding}em;
  font-size: 1.075em !important;
`
export const CardCaption = styled(Sidenote)`
  ${CardCaptionStyles};
`

export const CardHeader = styled.header`
  padding: ${props => props.theme.size.block.spacing / 4}em
    ${props => props.theme.size.block.spacing / 2}em;
  z-index: ${props => props.theme.layer.up};
  display: flex;
  justify-content: space-between;

  position: relative;
  box-shadow: 0 1px 1px
      ${props => props.theme.color.foreground(props.theme.opacity.least * 2)},
    0 0 0 1px
      ${props => props.theme.color.foreground(props.theme.opacity.least)};

  h3,
  input {
    ${props => props.theme.size.font.auto} ${props =>
  props.theme.typography.title.auto} white-space: nowrap;
    overflow: hidden;
    text-overflow: ellipsis;
    text-align: left;
    padding: 0;
  }
  h3::before,
  input::before {
    content: "${EMOJI.STAR}";
  }
  a {
    ${props => props.theme.size.font.auto} text-decoration:none;
    &:active {
      background: 0 0;
      color: ${props => props.theme.color.brand()};
    }
  }

`
