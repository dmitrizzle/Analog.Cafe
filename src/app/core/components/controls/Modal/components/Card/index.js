import React from "react"

import { Card } from "../../../Card"

// styles

// return
export const CardModal = props => {
  return (
    <Card
      {...props}
      style={{
        margin: "5vh auto 90.1vh" // 90.1vh is required for iOS to be scrollable
      }}
    />
  )
}
