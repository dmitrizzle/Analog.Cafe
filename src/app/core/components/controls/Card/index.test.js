import { shallow } from "enzyme"
import React from "react"

import Card from "./"

it("Render Card without crashing", () => {
  shallow(<Card />)
})
