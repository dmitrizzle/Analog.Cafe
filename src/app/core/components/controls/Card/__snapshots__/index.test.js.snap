// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render Card, matches snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <PureComponent />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <WithTheme(Component)
          buttons={undefined}
          error={undefined}
          noStar={undefined}
          stubborn={undefined}
          title={undefined}
/>,
        <Unknown
          image={undefined}
          text={undefined}
/>,
        undefined,
        undefined,
        undefined,
      ],
      "id": undefined,
      "style": undefined,
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "buttons": undefined,
          "error": undefined,
          "noStar": undefined,
          "stubborn": undefined,
          "title": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "image": undefined,
          "text": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      undefined,
      undefined,
      undefined,
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <WithTheme(Component)
            buttons={undefined}
            error={undefined}
            noStar={undefined}
            stubborn={undefined}
            title={undefined}
/>,
          <Unknown
            image={undefined}
            text={undefined}
/>,
          undefined,
          undefined,
          undefined,
        ],
        "id": undefined,
        "style": undefined,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "buttons": undefined,
            "error": undefined,
            "noStar": undefined,
            "stubborn": undefined,
            "title": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "image": undefined,
            "text": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        undefined,
        undefined,
        undefined,
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
